From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Wed, 22 May 2024 13:30:53 +0100
Subject: [PATCH] Ensure servicebus queue & topic have namespaces and RG names
 available

These are computed attributes for use in the servicebus mixins in Pulumi.

diff --git a/internal/services/servicebus/servicebus_queue_resource.go b/internal/services/servicebus/servicebus_queue_resource.go
index b85a172a9d..1d68962cb5 100644
--- a/internal/services/servicebus/servicebus_queue_resource.go
+++ b/internal/services/servicebus/servicebus_queue_resource.go
@@ -46,6 +46,8 @@ func resourceServiceBusQueue() *pluginsdk.Resource {
 	}
 }
 
+// FORK: @stack72: Ensured the ResourceGroupName and NamespaceName was available as a computed attribute for use
+// in the servicebus mixins in Pulumi
 func resourceServicebusQueueSchema() map[string]*pluginsdk.Schema {
 	return map[string]*pluginsdk.Schema{
 		"name": {
@@ -166,6 +168,16 @@ func resourceServicebusQueueSchema() map[string]*pluginsdk.Schema {
 			ForceNew: true,
 		},
 
+		"resource_group_name": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"namespace_name": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
 		"status": {
 			Type:     pluginsdk.TypeString,
 			Optional: true,
@@ -379,6 +391,8 @@ func resourceServiceBusQueueRead(d *pluginsdk.ResourceData, meta interface{}) er
 
 	d.Set("name", id.QueueName)
 	d.Set("namespace_id", namespaceId.ID())
+	d.Set("resource_group_name", id.ResourceGroupName)
+	d.Set("namespace_name", id.NamespaceName)
 
 	if model := resp.Model; model != nil {
 		if props := model.Properties; props != nil {
diff --git a/internal/services/servicebus/servicebus_topic_resource.go b/internal/services/servicebus/servicebus_topic_resource.go
index d106b6b017..c56e4bcd87 100644
--- a/internal/services/servicebus/servicebus_topic_resource.go
+++ b/internal/services/servicebus/servicebus_topic_resource.go
@@ -44,6 +44,8 @@ func resourceServiceBusTopic() *pluginsdk.Resource {
 	}
 }
 
+// FORK: @stack72: Ensured the ResourceGroupName and NamespaceName was available as a computed attribute for use
+// in the servicebus mixins in Pulumi
 func resourceServiceBusTopicSchema() map[string]*pluginsdk.Schema {
 	return map[string]*pluginsdk.Schema{
 		"name": {
@@ -134,6 +136,16 @@ func resourceServiceBusTopicSchema() map[string]*pluginsdk.Schema {
 			Type:     pluginsdk.TypeBool,
 			Optional: true,
 		},
+
+		"resource_group_name": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"namespace_name": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
 	}
 }
 
@@ -260,6 +272,10 @@ func resourceServiceBusTopicRead(d *pluginsdk.ResourceData, meta interface{}) er
 	d.Set("name", id.TopicName)
 	d.Set("namespace_id", topics.NewNamespaceID(id.SubscriptionId, id.ResourceGroupName, id.NamespaceName).ID())
 
+	// FORK: We must maintain these properties in order to keep Pulumi mixins working.
+	d.Set("namespace_name", id.NamespaceName)
+	d.Set("resource_group_name", id.ResourceGroupName)
+
 	if model := resp.Model; model != nil {
 		if props := model.Properties; props != nil {
 			status := ""
